controller_interface/include/semantic_components/force_torque_sensor.hpp

```diff
@@ -0,0 +1,169 @@
+   * e.g., 1D and 2D force load cells.
+   * For non existing axes interface is empty string, i.e., ("");
+   *
+   * The name should be in the following order:
+   *   force X, force Y, force Z, torque X, torque Y, torque Z.
+   */
+  ForceTorqueSensor(
+    const std::string & interface_force_x, const std::string & interface_force_y,
+    const std::string & interface_force_z, const std::string & interface_torque_x,
+    const std::string & interface_torque_y, const std::string & interface_torque_z
+  )
+  : SemanticComponentInterface("", 6)
+  {
+    auto check_and_add_interface =
+      [this](const std::string & interface_name, const int index)
+      {
+        if (!interface_name.empty()) {
+          interface_names_.emplace_back(interface_name);
+          existing_axes_[index] = true;
+        } else {
+          existing_axes_[index] = false;
+        }
+      };
+
+    check_and_add_interface(interface_force_x, 0);
+    check_and_add_interface(interface_force_y, 1);
+    check_and_add_interface(interface_force_z, 2);
+    check_and_add_interface(interface_torque_x, 3);
+    check_and_add_interface(interface_torque_y, 4);
+    check_and_add_interface(interface_torque_z, 5);
+
+    // Set default force and torque values to NaN
+    std::fill(forces_.begin(), forces_.end(), std::numeric_limits<double>::quiet_NaN());
+    std::fill(torques_.begin(), torques_.end(), std::numeric_limits<double>::quiet_NaN());
+  }
+
+  /// Return forces.
+  /**
+   * Return forces of a FTS.
+   *
+   * \return array of size 3 with force values.
+   */
+  std::array<double, 3> get_forces()
+  {
+    size_t interface_counter = 0;
+    for (size_t i = 0; i < 3; ++i) {
+      if (existing_axes_[i]) {
+        forces_[i] = state_interfaces_[interface_counter].get().get_value();
+        ++interface_counter;
+      }
+    }
+    return forces_;
+  }
+
+  /// Return torque.
+  /**
+   * Return torques of a FTS.
+   *
+   * \return array of size 3 with torque values.
+   */
+  std::array<double, 3> get_torques()
+  {
+    // find out how many force interfaces are being used
+    // torque interfaces will be found from the next index onward
+    auto torque_interface_counter = std::count(
+      existing_axes_.begin(),
+      existing_axes_.begin() + 3, true);
+
+    for (size_t i = 3; i < 6; ++i) {
+      if (existing_axes_[i]) {
+        torques_[i - 3] = state_interfaces_[torque_interface_counter].get().get_value();
+        ++torque_interface_counter;
+      }
+    }
+    return torques_;
+  }
+
+  /// Return Wrench message with forces and torques.
+  /**
+   * Constructs and return a wrench message from the current values.
+   * The method assumes that the interface names on the construction are in the following order:
+   *   force X, force Y, force Z, torque X, torque Y, torque Z.
+   *
+   * \return wrench message from values;
+   */
+  geometry_msgs::msg::Wrench get_values_as_message()
```

Comment:
These sound like something that'd be great to have in semantic component as a base function
```
template <typename MessageReturnType>
class SemanticComponentInterface
.............


bool get_values_as_message(MessageReturnType& message)
{ return false; }
```
....
```
class ForceTorqueSensor : public SemanticComponentInterface<geometry_msgs::msg::Wrench>
{
...
}
```

Reply:
@bmagyar , What should be the validation for `false` of `MessageReturnType` in `get_values_as_message` of `ForceTorqueSensor` ?

Reply:
Added `get_values_as_message` as base function in `semantic_component`

Reply:
`return false` just signals to whoever calls it that they should not have any expectations of the outcome of the function call but if you think it's better to force people to implement the function you can make it pure abstract as well with `= 0;`

Reply:
Got your point. But I would not go forward with the pure virtual as it will 'force' the coder to write a dummy placeholder function. So, just to keep everything simple, I think it is better to keep all the arrangement as it is. :)